internal/analytics/client/sync/reporters/pixel.go

```diff
@@ -7,32 +7,49 @@ import (
 	"os"
 
 	"github.com/ActiveState/cli/internal/analytics/dimensions"
+	"github.com/ActiveState/cli/internal/config"
 	"github.com/ActiveState/cli/internal/constants"
 	"github.com/ActiveState/cli/internal/errs"
+
+	configMediator "github.com/ActiveState/cli/internal/mediators/config"
 )
 
 type PixelReporter struct {
-	url string
+	cfg *config.Instance
 }
 
-func NewPixelReporter() *PixelReporter {
-	var pixelUrl string
-
-	// Attempt to get the value for the pixel URL from the environment.  Fall back to default if that fails
-	if pixelUrl = os.Getenv(constants.AnalyticsPixelOverrideEnv); pixelUrl == "" {
-		pixelUrl = constants.DefaultAnalyticsPixel
-	}
-	return &PixelReporter{pixelUrl}
+func NewPixelReporter(cfg *config.Instance) *PixelReporter {
+	reporter := &PixelReporter{cfg: cfg}
+	configMediator.AddListener(constants.AnalyticsPixelOverrideConfig, func() {
+		reporter.cfg = cfg
+	})
+	return reporter
 }
 
 func (r *PixelReporter) ID() string {
 	return "PixelReporter"
 }
 
 func (r *PixelReporter) Event(category, action, source, label string, d *dimensions.Values) error {
-	pixelURL, err := url.Parse(r.url)
+	var (
+		pixelUrl string
+
+		envUrl = os.Getenv(constants.AnalyticsPixelOverrideEnv)
+		cfgUrl = r.cfg.GetString(constants.AnalyticsPixelOverrideConfig)
+	)
+
+	switch {
+	case envUrl != "":
+		pixelUrl = envUrl
+	case cfgUrl != "":
+		pixelUrl = cfgUrl
+	default:
+		pixelUrl = constants.DefaultAnalyticsPixel
+	}
+
+	pixelURL, err := url.Parse(pixelUrl)
```

Comment:
Suggest making this its own function that gets invoked both in the constructor and when the config value changes. You could make that a private method on PixelReporter which then also updates the url.